sherpa/plot/backends.py

```diff
@@ -0,0 +1,995 @@
+           as a template only.
+
+        Parameters
+        ----------
+        x : array-like or scalar number
+            x values
+        y : array-like or scalar number
+            y values, same dimension as `x`.
+        {kwargs}
+        """
+        pass
+
+    @add_kwargs_to_doc(kwargs_doc)
+    @translate_args
+    def histo(self, xlo, xhi, y, *,
+              yerr=None,
+              title=None, xlabel=None, ylabel=None,
+              overplot=False, clearwindow=True,
+              xlog=False, ylog=False,
+              label=None,
+              xerrorbars=False,
+              yerrorbars=False,
+              color=None,
+              linestyle='solid',
+              linewidth=None,
+              marker='None',
+              alpha=None,
+              markerfacecolor=None,
+              markersize=None,
+              ecolor=None,
+              capsize=None,
+              # barsabove=False,
+              **kwargs):
+        """Draw histogram data.
+
+        The histogram is drawn as horizontal lines connecting the
+        start and end points of each bin, with vertical lines connecting
+        consecutive bins. Non-consecutive bins are drawn with a
+        (NaN, NaN) between them so no line is drawn connecting them.
+
+        Points are drawn at the middle of the bin, along with any
+        error values.
+
+        .. warning::
+           This backend is a non-functional dummy. The documentation is provided
+           as a template only.
+
+        Parameters
+        ----------
+        x0 : array-like or scalar number
+            lower bin boundary values
+        x1 : array-like or scalar number
+            upper bin boundary values
+        y : array-like or scalar number
+            y values, same dimension as `x0`.
+        {kwargs}
+        """
+        pass
+
+    @add_kwargs_to_doc(kwargs_doc)
+    @translate_args
+    def contour(self, x0, x1, y, *,
+                levels=None,
+                title=None, xlabel=None, ylabel=None,
+                overcontour=False, clearwindow=True,
+                xlog=False, ylog=False,
+                label=None,
+                colors=None,
+                linestyles='solid',
+                linewidths=None,
+                alpha=None,
+                **kwargs):
+        """Draw 2D contour data.
+
+        .. warning::
+           This backend is a non-functional dummy. The documentation is provided
+           as a template only.
+
+        Parameters
+        ----------
+        x0 : array-like
+            independent axis in the first dimenation (on regular grid, flattened)
+        x1 : array-like
+            independent axis in the second dimenation (on regular grid, flattened)
+        y : array-like
+            dependent axis (i.e. image values) (on regular grid, flattened)
+        {kwargs}
+        """
+        pass
+
+    @translate_args
+    def image(self, x0, x1, y, *,
```

Comment:
I think we could remove this from this PR as we don't seem to exercise this, and I think there's likely to be a bunch of bike-shedding on the image support.

Reply:
We could, if you want to bike-shed. My hope was to get a minimal version in here that's hopefully not controversial [*] and we can always add to it later and backend are allowed to accept any other kwargs. The `x0, x1,y` is pretty obvious and all other arguments are keyword only, so we can add more later. 

How about a minimal
```
    def image(self, x0, x1, y, *,
              title=None, xlabel=None, ylabel=None,
              overplot=False, clearwindow=True, **kwargs)
```
for now as a compromise for now? I think there is a benefit to have at least a minimal version here, because I speced it in #1375 and I don't want to drag this out too far.

Reply:
I still don't see the need for this in this PR.